src/Microsoft.AspNetCore.Mvc.Core/Routing/MvcEndpointDataSource.cs

```diff
@@ -184,21 +157,18 @@ private void UpdateEndpoints()
                         }
 
                         var attributeRoutePattern = RoutePatternFactory.Parse(action.AttributeRouteInfo.Template);
+                        var updatedRoutePattern = _routePatternTransformer.SubstituteRequiredValues(attributeRoutePattern, action.RouteValues);
```

Comment:
@rynowak I think this will fail. `SubstituteRequiredValuesCore` combines the matching that convention routes require with substitution. Attribute routes don't need to match.

Reply:
I don't really understand this comment? Maybe an example would help? Are you doing this for attribute routes to support processing parameters in them?

Reply:
Yes, required parameters in attributes are supported with attribute routes, e.g. `[Route("api/{controller}/{"action=Index")]`

Reply:
https://github.com/aspnet/Mvc/blob/master/test/WebSites/RoutingWebSite/Controllers/EndpointRoutingController.cs